torch/utils/mkldnn.py

```diff
@@ -63,26 +58,51 @@ def __getstate__(self):
         return (self.weight.to_dense(), self.bias.to_dense(), self.training)
 
     @torch.jit.script_method
-    def __setstate__(self, state):
-        self.weight = torch._C._nn.mkldnn_reorder_conv2d_weight(
-            state[0].to_mkldnn(),
+    def forward(self, x):
+        return torch.mkldnn_convolution(
+            x,
+            self.weight,
+            self.bias,
             self.padding,
             self.stride,
             self.dilation,
             self.groups)
+
+
+class MkldnnConv1d(_MkldnnConvNd):
+    def __init__(self, dense_module):
+        super(MkldnnConv1d, self).__init__(dense_module)
+
+        self.register_buffer('weight', dense_module.weight.to_mkldnn())
+
+    @torch.jit.script_method
+    def __setstate__(self, state):
+        self.weight = state[0].to_mkldnn()
         self.bias = state[1].to_mkldnn()
         self.training = state[2]
 
-    @torch.jit.script_method
-    def forward(self, x):
-        return torch.conv2d(
-            x,
-            self.weight,
-            self.bias,
+
+class MkldnnConv2d(_MkldnnConvNd):
+    def __init__(self, dense_module):
+        super(MkldnnConv2d, self).__init__(dense_module)
+
+        self.register_buffer('weight', torch._C._nn.mkldnn_reorder_conv2d_weight(
```

Comment:
Out of curiosity why are these buffers and not parameters? They are not expected to be returned when you do `model.parameter()` ?